packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2734,91 +2737,87 @@ export class ContainerRuntime
 		messageWithContext: MessageWithContext,
 		localOpMetadata: unknown,
 	): void {
-		// TODO: destructure message and modernRuntimeMessage once using typescript 5.2.2+
-		const { local } = messageWithContext;
-		switch (messageWithContext.message.type) {
-			case ContainerMessageType.Attach:
-			case ContainerMessageType.Alias:
-			case ContainerMessageType.FluidDataStoreOp:
-				this.channelCollection.process(messageWithContext.message, local, localOpMetadata);
-				break;
-			case ContainerMessageType.BlobAttach:
-				this.blobManager.processBlobAttachOp(messageWithContext.message, local);
-				break;
-			case ContainerMessageType.IdAllocation:
-				// Don't re-finalize the range if we're processing a "savedOp" in
-				// stashed ops flow. The compressor is stashed with these ops already processed.
-				// That said, in idCompressorMode === "delayed", we might not serialize ID compressor, and
-				// thus we need to process all the ops.
-				if (!(this.skipSavedCompressorOps && messageWithContext.savedOp === true)) {
-					const range = messageWithContext.message.contents;
-					// Some other client turned on the id compressor. If we have not turned it on,
-					// put it in a pending queue and delay finalization.
-					if (this._idCompressor === undefined) {
-						assert(
-							this.idCompressorMode !== undefined,
-							0x93c /* id compressor should be enabled */,
-						);
-						this.pendingIdCompressorOps.push(range);
-					} else {
-						assert(
-							this.pendingIdCompressorOps.length === 0,
-							0x979 /* there should be no pending ops! */,
-						);
-						this._idCompressor.finalizeCreationRange(range);
+		if (messageWithContext.isModernRuntimeMessage) {
```

Comment:
Confused - this looks redundant with the if check at the call site?

Reply:
I agree, added it in case there were other places in future where this could be called from + for avoiding the error message of incompatible type in the default condition. 

I will remove it, looks unecessary.

Reply:
There may be value in keeping it if `validateAndProcessRuntimeMessage` is never supposed to deal with `ContainerMessageType.Operation` and the code in `default` doesn't really add any value (in logs, telemetry, debugging of future incidents) if it runs for that type of message.

If you do decide to keep it, consider early-returns as opposed to large if-blocks. Helps readability. 

`if (messageWithContext.isModernRuntimeMessage) {
  return; 
  } 

Switch(message.type)`